pkg/instrumentation/sdk.go

```diff
@@ -295,18 +295,18 @@ func (i *sdkInjector) injectCommonSDKConfig(ctx context.Context, otelinst v1alph
 		}
 	}
 
-	// Some attributes might be empty, we should get them via k8s downward API
-	if resourceMap[string(semconv.K8SPodNameKey)] == "" {
-		container.Env = append(container.Env, corev1.EnvVar{
-			Name: constants.EnvPodName,
-			ValueFrom: &corev1.EnvVarSource{
-				FieldRef: &corev1.ObjectFieldSelector{
-					FieldPath: "metadata.name",
-				},
+	// Always retrieve the pod name from the Downward API. Ensure that the OTEL_RESOURCE_ATTRIBUTES_POD_NAME env exists.
+	container.Env = append(container.Env, corev1.EnvVar{
+		Name: constants.EnvPodName,
+		ValueFrom: &corev1.EnvVarSource{
+			FieldRef: &corev1.ObjectFieldSelector{
+				FieldPath: "metadata.name",
 			},
-		})
-		resourceMap[string(semconv.K8SPodNameKey)] = fmt.Sprintf("$(%s)", constants.EnvPodName)
-	}
+		},
+	})
+	resourceMap[string(semconv.K8SPodNameKey)] = fmt.Sprintf("$(%s)", constants.EnvPodName)
```

Comment:
We should also stop setting this in `createResourceMap`.

Reply:
https://github.com/open-telemetry/opentelemetry-operator/blob/934c1d8ef90dbf0d570c52e115148c09c63d3959/pkg/instrumentation/sdk.go#L416-L420

We can retain this, because other places are using it.

Reply:
also `nginx` and `httpd`